src/Test/QuickCheck/Gen.purs

```diff
@@ -160,9 +160,13 @@ runGen = runState
 evalGen :: forall a. Gen a -> GenState -> a
 evalGen = evalState
 
--- | Sample a random generator
-sample :: forall r a. Seed -> Size -> Gen a -> Array a
-sample seed sz g = evalGen (vectorOf sz g) { newSeed: seed, size: sz }
+-- | Sample a random generator, using the given seed and producing a sample
+-- | containing the given number of values.
+sample :: forall r a. Seed -> Int -> Gen a -> Array a
+sample seed n g = evalGen sampleGen { newSeed: seed, size: 0 }
+  where
+  sampleGen = traverse (flip resize g) sizes
+  sizes = range 0 (n - 1) <#> (*2)
```

Comment:
This can result in very large sizes. Perhaps something like this would work?

```purescript
sampleGen = traverse (flip resize g) sizes
  sizes = range 0 (n - 1) <#> f
    where
    f n | n >= 20 = 20
    f n = n + 2
```

In Haskell Quickcheck's `sample'` it goes up from 0 in twos and is capped at 20. I think this produces 11 results?

```haskell
sample' g = generate (sequence [ resize n g | n <- [0,2..20] ])
```

It might be a good idea to include a size of one as well to make sure we get a singleton array.